androidlibrary_lib/src/test/java/org/opendatakit/utilities/DateUtilsTest.java

```diff
@@ -46,4 +50,66 @@ public void testDateInterpretation() {
     assertEquals(expected, value.substring(0,expected.length()));
   }
 
+  @Test
+  public void testNowInput() {
+    TimeZone tz = TimeZone.getTimeZone(TimeZone.getAvailableIDs()[0]);
+    DateUtils util = new DateUtils(Locale.US, tz);
+
+    String value = util.validifyDateValue("now");
+    assertNotNull(value);
+
+    DateTime now = new DateTime();
+    String expectedFormattedDate = util.formatDateTimeForDb(now);
+
+    // This is to take the slight delay when checking for output into consideration
+    int periodIndex = expectedFormattedDate.indexOf('.');
+    assertTrue(value.startsWith(expectedFormattedDate.substring(0, periodIndex + 1)));
+  }
+
+  @Test
+  public void testTimeAddition() {
```

Comment:
You can make your test names more descriptive to enhance readability. You could follow a structure like `given_when_then` or use the pattern `subjectUnderTest_actionOrInput_expectedResult`